shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -77,6 +87,53 @@ typedef NS_ENUM(NSUInteger, FlutterTextAffinity) {
   return flutter::TextRange([base unsignedLongValue], [extent unsignedLongValue]);
 }
 
+// Returns the autofill hint content type, if specified; otherwise nil.
+static NSString* GetAutofillContentType(NSDictionary* configuration) {
+  NSDictionary* autofill = configuration[kAutofillProperties];
+  NSArray<NSString*>* hints = autofill[kAutofillHints];
+  return hints.count > 0 ? hints[0] : nil;
+}
+
+// Returns YES if configuration describes a field for which autocomplete should be enabled for
+// the specified TextInputConfiguration. Autocomplete is enabled by default, but will be disabled
+// if the field is password-related, or if the autofill hint is the empty string.
+static BOOL EnableAutocompleteForTextInputConfiguration(NSDictionary* configuration) {
+  // Disable if obscureText is set.
+  if ([configuration[kSecureTextEntry] boolValue]) {
+    return NO;
+  }
+
+  // Disable if autofill properties indicate a username/password.
+  NSString* contentType = GetAutofillContentType(configuration);
```

Comment:
Consider checking for `nil` here if there are no autofill content types.

Reply:
In Obj-C, sending a message to nil always results in nil, so it's more idiomatic just to go ahead and fire the message. In this case, the ensuing checks will all be false in that case.